### PR TITLE
chore: updated the docs for this [PR](https://github.com/HaoboGu/rmk-…

### PR DESCRIPTION
**Pull request by fhanrath**
This is an update to the docs for [this](https://github.com/HaoboGu/rmk-template/pull/11) pull request.

### CODE REVIEW

File: examples/use_rust/stm32f4/Makefile.toml [hunk 1]
```diff
@@ -14,14 +14,18 @@ args = ["objcopy", "--release", "--", "-O", "ihex", "rmk.hex"]
...
+    "--input-path",
     "rmk.hex",
-    "-c",
-    "-f",
-    "0x2dc309c5",                  # Use your chip family ID in the uf2conv.py file as -f argument
-    "-o",
+    "--family",
     "rmk.uf2",
+    "--family",
+    "stm32f411xe",
```
Comment (HaoboGu) [inline examples/use_rust/stm32f4/Makefile.toml]: Should the family be `stm32f4`? I see `use_config/stm32f4/Makefile.toml` have this.

File: examples/use_config/nrf52832_ble/Makefile.toml [hunk 2]
```diff
@@ -14,14 +14,18 @@ args = ["objcopy", "--release", "--", "-O", "ihex", "rmk.hex"]
...
+    "--input-path",
     "rmk.hex",
-    "-c",
-    "-f",
-    "0x72721d4e",                  # Use your chip family ID in the uf2conv.py file as -f argument
-    "-o",
+    "--output-path",
     "rmk.uf2",
+    "--family",
+    "nrf52832xxaa",
```
Comment (HaoboGu) [inline examples/use_config/nrf52832_ble/Makefile.toml]:
```
[cargo-make] INFO - Execute Command: "cargo" "hex-to-uf2" "--input-path" "rmk.hex" "--output-path" "rmk.uf2" "--family" "nrf52832xxaa"
error: invalid value 'nrf52832xxaa' for '--family <FAMILY>'

  tip: a similar value exists: 'nrf52832xx-aa'
```

Wrong family name?

**Comment by HaoboGu** [comment]
Thanks a lot! There are some minor issues, we can get it merged after fixing them.

File: docs/src/user_guide/2-2_local_compilation.md [hunk 3]
```diff
@@ -153,14 +151,14 @@ If you encountered any problems when compiling the firmware, please report it [h
 
 By default, Rust firmware is an ELF file, so we have to do some extra steps converting it to uf2 format.
 
-RMK uses [cargo-make](https://github.com/sagiegurari/cargo-make) to automate the uf2 firmware generation. Generating uf2 firmware also requires you have `python` command available.
+RMK uses [cargo-make](https://github.com/sagiegurari/cargo-make) to automate the uf2 firmware generation.
 
-Then you should make sure the chip family argument(aka argument after -f) in `Makefile.toml` is correct. You can get your chip's family ID in `scripts/uf2conv.py`.
+Then you should make sure the chip family argument(aka argument after --family) in `Makefile.toml` is correct. You can get your chip's family [here](https://git.sr.ht/~fenris/hex-to-uf2/tree/main/item/hex_to_uf2/src/families.rs#L7).
```
Comment (HaoboGu) [inline docs/src/user_guide/2-2_local_compilation.md]: Could you publish it on GitHub? A GitHub link is better

**Comment by fhanrath** [comment]
Alright, what issues are you referring to specifically?

**Comment by HaoboGu** [comment]
> Alright, what issues are you referring to specifically?

Could you see the pending comments?

**Comment by fhanrath** [comment]
> Could you see the pending comments?
> 
> I can see it in the PR history

I'm not able to see anything like that. Only the comments on this page.
Other than that, I only see the "Commits", "Checks" and "Files changed" tab.

It looks like your Reviews are pending. I'm not too familiar with GitHub, but do you maybe need to commit or finish your review before others will be able to see it?

**Comment by fhanrath** [comment]
> image

Regarding the first one:
I've compared the family ID that was used in the example before (`0x2dc309c5`) to the [list of families](https://git.sr.ht/~fenris/hex-to-uf2/tree/main/item/hex_to_uf2/src/families.rs#L142) and their mapping to IDS and found that `stm32f411x-e` maps to the correct ID.

Regarding the second one:
There are some families where Clap's ValueEnum injects a `-`between letters. Didn't notice this before, so I'll update the examples.

**Comment by HaoboGu** [comment]
oh I forgot to submit the review, sorry for that!

**Comment by HaoboGu** [comment]
> I've compared the family ID that was used in the example before (`0x2dc309c5`) to the [list of families](https://git.sr.ht/~fenris/hex-to-uf2/tree/main/item/hex_to_uf2/src/families.rs#L142) and their mapping to IDS and found that `stm32f411x-e` maps to the correct ID. The list itself I constructed from the [original python script](https://github.com/HaoboGu/rmk/blob/main/scripts/uf2conv.py#L321).

Got it. let's update it to `stm32f4` now. There're some inconsistencies in the previous code.

File: examples/use_rust/stm32f4/Makefile.toml (hunk 1 above)
Comment (fhanrath) [reply examples/use_rust/stm32f4/Makefile.toml]: fixed

File: examples/use_config/nrf52832_ble/Makefile.toml (hunk 2 above)
Comment (fhanrath) [reply examples/use_config/nrf52832_ble/Makefile.toml]: fixed

File: docs/src/user_guide/2-2_local_compilation.md (hunk 3 above)
Comment (fhanrath) [reply docs/src/user_guide/2-2_local_compilation.md]: adjusted

File: examples/use_rust/stm32f4/Makefile.toml [hunk 4]
```diff
@@ -14,14 +14,18 @@ args = ["objcopy", "--release", "--", "-O", "ihex", "rmk.hex"]
...
 args = [
-    "../../../scripts/uf2conv.py",
+    "hex-to-uf2",
+    "--input-path",
     "rmk.hex",
-    "-c",
-    "-f",
-    "0x2dc309c5",                  # Use your chip family ID in the uf2conv.py file as -f argument
-    "-o",
+    "--family",
```
Comment (HaoboGu) [inline examples/use_rust/stm32f4/Makefile.toml]: wrong argument

**Comment by HaoboGu** [comment]
pls fix the ci error then I can merge it

**Comment by HaoboGu** [comment]
Great work! I'm merging it, thanks a lot!

**Comment by HaoboGu** [comment]
There's something we can do to improve it, for example, releasing the cargo-hex-to-uf2 binary so that we can use `cargo binstall` to speedup the CI. Anyway, that's the future work.

**Pull request merged**